lib/datadog/profiling/component.rb

```diff
@@ -110,6 +116,52 @@ def self.build_profiler_component(settings:, agent_settings:, optional_tracer:)
         end
       end
 
+      private_class_method def self.get_allocation_sample_every(settings)
+        allocation_sample_rate = settings.profiling.advanced.experimental_allocation_sample_rate
+
+        if allocation_sample_rate <= 0
+          raise("Allocation sample rate must be a positive integer. Was #{allocation_sample_rate}")
+        end
```

Comment:
Minor: Here and below, consider raising an `ArgumentError`, as it's a bit more idiomatic?